CKSIncrementalStore/CKSIncrementalStoreSyncOperation.swift

```diff
@@ -302,7 +302,7 @@ class CKSIncrementalStoreSyncOperation: NSOperation {
             
             if operationError == nil
             {
-                self.saveServerChangeToken(serverChangeToken: serverChangeToken)
+                self.saveServerChangeToken(serverChangeToken: serverChangeToken) // TODO: saving this here means that if the local save fails or is interrupted then this whole change set will be lost. Would be better to do this after saving locally.
```

Comment:
This is a great suggestion. Will try out to incorporate this in code Thanks man :)